docs(aws-machine-image): clarify that currentImageName is updated by Renovate

josiblair opened the pull request:
## Description

The example regex pattern incorrectly captures `currentImageName` into the `currentDigest` group, causing Renovate to update this comment field. The `currentImageName` field should remain static (typically "unknown") and not be modified during updates.

Remove the named capture group `<currentDigest>` from the `currentImageName` pattern so the comment line is preserved as-is while only the AMI ID is updated.

<!-- If this is your first pull request: sign the CLA with this GitHub app: https://cla-assistant.io/renovatebot/renovate -->
<!-- Make sure the `Allow edits and access to secrets by maintainers` checkbox is checked on this pull request. -->
<!-- Please read https://github.com/renovatebot/renovate/blob/main/.github/contributing.md before you create your pull request.-->

## Changes

Updated the example regex pattern in the AWS Machine Image datasource documentation to remove the `<currentDigest>` named capture group from the `currentImageName` line. This prevents Renovate from updating the `# currentImageName=unknown` comment, which should remain static.

Before: `(.*currentImageName=(?<currentDigest>.*?)\n)?`
After: `(.*currentImageName=.*?\n)?`

## Context

Please select one of the below:

- [ ] This closes an existing Issue: #
- [x] This doesn't close an Issue, but I accept the risk that this PR may be closed if maintainers disagree with its opening or implementation

## AI assistance disclosure

Did you use AI tools to create any part of this pull request?

- [ ] No — I did not use AI for this contribution.
- [ ] Yes — minimal assistance (e.g., IDE autocomplete, small code completions, grammar fixes).
- [x] Yes — substantive assistance (AI generated non‑trivial portions of code, tests, or documentation).
- [ ] Yes — other (please describe):

AI was used to help identify the issue in the regex pattern.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

josiblair committed:
docs(aws-machine-image): fix regex to preserve currentImageName comment

The example regex pattern incorrectly captures `currentImageName` into the `currentDigest` group, causing Renovate to update this comment field. The `currentImageName` field should remain static (typically "unknown") and not be modified during updates.

Remove the named capture group `<currentDigest>` from the `currentImageName` pattern so the comment line is preserved as-is while only the AMI ID is updated.

josiblair commented:
It's not public. It's on a self-hosted instance of GitLab. Here are some screenshots of before/after the regex change. 
<img width="1586" height="766" alt="Screenshot 2025-10-08 at 9 43 15 AM" src="https://github.com/user-attachments/assets/acf4be62-c5f7-4f71-ae58-040e4148304e" />
<img width="1226" height="350" alt="Screenshot 2025-10-08 at 10 37 29 AM" src="https://github.com/user-attachments/assets/bf1a9b28-85e3-4eb9-878f-bc781d2be21e" />
<img width="1236" height="705" alt="Screenshot 2025-10-08 at 10 38 43 AM" src="https://github.com/user-attachments/assets/1bce0509-a518-4e31-a6c2-b274edd7c959" />

jamietanna reviewed (commented):
So one thing this will lead to is the removal of the `currentDigest` - is that intended? I notice that in the "before" photo, we're updating the `currentImageName` to what looks like a useful new value?

josiblair committed:
chore: revert quote changes

RahulGautamSingh reviewed (commented):
It is supposed to track the digest. But in the examples we haven't used any digest. So, if you could update one of them to digest that'd be great, given that digests for AMI images exist like for other manager packages

josiblair committed:
docs(aws-machine-image): clarify that currentImageName is updated by renovate

Replace `currentImageName=unknown` with realistic AMI names in all examples to accurately reflect that Renovate updates this field to track the actual AMI name. Add note explaining this behavior.

RahulGautamSingh commented on lib/modules/datasource/aws-machine-image/readme.md:
```diff
@@ -93,46 +93,51 @@ Or as JSON:
 
+**Note about `currentImageName`:**
+
+The optional `currentImageName` comment is automatically updated by Renovate to track the actual AMI name corresponding to the AMI ID. This provides human-readable context about which image version is being used. When Renovate finds a newer AMI, it will update both the AMI ID and the `currentImageName` comment.
```
One sentence per line

```suggestion
The optional `currentImageName` comment is automatically updated by Renovate to track the actual AMI name corresponding to the AMI ID.
This provides human-readable context about which image version is being used.
When Renovate finds a newer AMI, it will update both the AMI ID and the `currentImageName` comment.
```

RahulGautamSingh reviewed (commented):
Else, LGTM

jamietanna reviewed (commented):
One comment from Rahul, and the other change I'd like to see is an example _without_ the `currentImageName` - I see from the regex (and our improved docs) that it shouldn't be required, but it'd be good to have that confirmed, as part of the docs 🙏  

Otherwise LGTM, thanks for the back-and-forth!

josiblair committed:
chore: update structure

Co-authored-by: RahulGautamSingh <rahultesnik@gmail.com>

josiblair committed:
docs(aws-machine-image): add examples without currentImageName comment

josiblair committed:
chore: add typescript example

jamietanna reviewed (approved):
Thank you!
